source/agora/node/Ledger.d

```diff
@@ -229,6 +235,57 @@ public class Ledger
         else
             return null;
     }
+
+    /***************************************************************************
+
+        Write a block to the ledger file on disk.
+
+        Params:
+            fd = Instance of LedgerFile
+            block = Type of struct to writeFile
+
+    ***************************************************************************/
+
+    public void writeBlockFile (ref File fd, ref Block block) @trusted
+    {
+        const string dir = this.data_dir;
+        size_t block_size = 0;
+
+         // Make sure we're in the right directory
+        assert(dir.exists);
+        if (!std.file.exists(dir) || !std.file.isDir(dir))
+        std.file.mkdir(dir);
+
+         scope SerializeDg dg = (scope const (ubyte[]) data) @safe
+        {
+            try
+            {
+                fd.rawWrite(data);
+                block_size += data.length;
+            }
+            catch (ErrnoException ex)
+            {
+                //logError("writeBlockFile: %s", ex.message);
+            }
+        };
+        serializePart(block, dg);
+
+         string file_path = format("%s/%s", dir, "index.dat");
+        scope File index_fd = File(file_path, "a");
```

Comment:
`scope` has no meaning here.

It is only useful when you do `scope object = new Class()`, which will allocate the class no the stack.

Reply:
Fixed